### DISCUSSION TITLE
Elements removal animation

### DISCUSSION THREAD

**Discussion by yexiang92** [pyvista/pyvista#3712] [Q&A] [answered]
Hello everyone, I am a deep fan of pyvista. I recently wanted to implement animations about cell removal, but pyvista does not seem to provide this function.

The specific idea is that during the animation, I have some cells removed, and the related coordinate points and cells should be deleted at this time. In other words, at each step, I should update the coordinate points and cells of different sizes, for pyvista It seems that this function is not currently available.
Can pyvista increase this function? Thanks.

For example, an animation like the one below.

![NodalDisp](https://user-images.githubusercontent.com/112955856/207288436-adc43b38-3a22-4c8b-98de-6e2a05af9386.gif)

**Comment by banesullivan** [comment, marked as answer]
The `remove_cells` method might be what you're after: https://docs.pyvista.org/api/core/_autosummary/pyvista.PointSet.remove_cells.html#remove-cells

Here is an example:

```py
import pyvista as pv
import numpy as np
from pyvista import examples

mesh = examples.load_hexbeam()

plotter = pv.Plotter(off_screen=True, notebook=False)
# Open a movie file
plotter.open_movie('movie.mp4')

# Add initial mesh
plotter.add_mesh(mesh, color=True, show_edges=True)
# Add outline for shrinking reference
plotter.add_mesh(mesh.outline_corners())

plotter.show(auto_close=False)  # only necessary for an off-screen movie

# Run through each frame
plotter.write_frame()  # write initial data

# Update scalars on each frame
for i in range(90):
    if i%10 == 0:  # To slow down frame rate
        ids = np.array([mesh.n_cells]*4) - range(1, 5)
        mesh.remove_cells(ids, inplace=True)
    plotter.write_frame()  # Write this frame

# Be sure to close the plotter when finished
plotter.close()
```

https://user-images.githubusercontent.com/22067021/207734437-f4e1d4dd-631e-45f2-9992-82527694697d.mp4

**Reply by yexiang92** [reply]
Thank you very much for your reply, it looks very cool.
